modules/cij/runner.py

```diff
@@ -181,6 +181,69 @@ def tsuites_from_dicts(tsuites_dicts) -> List[TestSuite]:
         return tsuites
 
 
+@dataclasses.dataclass
+class TestPlan:
+    """
+    TestPlans contain information required to execute a testplan.
+    """
+    # pylint: disable=too-many-instance-attributes
+
+    fpath_orig: str = ""
+    fpath: str = ""
+    fname: str = ""
+    name: str = ""
+    ident: str = ""
+
+    res_root: str = ""
+    aux_root: str = ""
+    aux_list: list = dataclasses.field(default_factory=list)
+
+    stamp: dict = dataclasses.field(default_factory=lambda: {
+        "begin": None,
+        "end": None
+    })
+    status: str = Status.Unkn
+    status_preq: str = Status.Unkn
+    wallc: Optional[float] = None
+    log_content: str = ""
+
+    evars: dict = dataclasses.field(default_factory=dict)
+    hooks: dict = dataclasses.field(default_factory=lambda: {
+        "enter": [],
+        "exit": []
+    })
+    hnames: list = dataclasses.field(default_factory=list)
+
+    testsuites: list = dataclasses.field(default_factory=list)
+
+    @staticmethod
+    def from_dict(tplan_dict) -> TestPlan:
+        """
+        Deserialize a tplan dict into TestPlan
+        """
+        complex_keys = {
+            'testsuites': TestSuite.tsuites_from_dicts,
+            'hooks': Hook.hooks_from_dict,
+        }
+
+        tplan = TestPlan()
+
+        fields = _get_dataclass_fields(TestPlan) & set(tplan_dict)
+        for key in fields:
+            deserialize = complex_keys.get(key, lambda x: x)
+            setattr(tplan, key, deserialize(tplan_dict[key]))
+
+        return tplan
+
+    @staticmethod
+    def from_lod(lod) -> List[TestPlan]:
```

Comment:
`lod` means list of dicts, right? It took me a second to realize -- maybe it would be easier if we added a type annotation, e.g. `lod: Dict[str, Any]` (or a more specific type instead of Any, if that's possible)

Reply:
This name sticks out from the naming of the other methods that does the same on our other classes. It might make sense to change all of them to this shorter naming `from_lod`, otherwise I think this one should be renamed `tplans_from_dicts`

Reply:
I guess the type: ``lod: List[dict]`` would fit?
Regarding the naming, yeah, I'll switch to match what is there. Then a cleanup can be done later for changing naming overall.

Reply:
Ah yeah, sorry, I completely messed that type signature up!
I think we might get a linter error if we don't define what's in the dict, but give it a try

Reply:
mypy does not catch violations regardless of the annotation :/ Seem like it is because the function is applied through the "complex-key"  indirection, possibly just too much runtime polymorphism for mypy :)